library/bignum_mod.h

```diff
@@ -75,16 +74,30 @@ typedef struct {
 
 /** Setup a residue structure.
  *
- * \param[out] r    The address of residue to setup. The size is determined by
- *                  \p m.
- *                  (In particular, it must have at least as many limbs as the
- *                  modulus \p m.)
+ * The residue will be set up with the \p p buffer \p m modulus.
+ *
+ * The memory pointed by \p p will be used by the resulting residue structure.
+ * The value at the pointed memory will be the initial value of \p r and must
+ * hold a value that is less than the modulus. This value will be used as it is
+ * and interpreted according to the value of the `m->int_rep` field.
+ *
+ * The modulus \p m will be the modulus associated with \p r. The residue \p r
+ * should only be used in operations where the modulus is \p m or a modulus
+ * equivalent to \p m (in the sense that all their fields or memory pointed by
+ * their fields hold the same value).
+ *
+ * \param[out] r    The address of residue to setup. The resulting structure's
+ *                  size is determined by \p m.
  * \param[in] m     The address of the modulus related to \p r.
  * \param[in] p     The address of the limb array storing the value of \p r.
  *                  The memory pointed to by \p p will be used by \p r and must
  *                  not be modified in any way until after
- *                  mbedtls_mpi_mod_residue_release() is called.
- * \param p_limbs   The number of limbs of \p p.
+ *                  mbedtls_mpi_mod_residue_release() is called. The data
+ *                  pointed by \p p should be less than the modulus (the value
+ *                  pointed by `m->p`) and already in the representation
+ *                  indicated by `m->int_rep`.
+ * \param p_limbs   The number of limbs of \p p. It must have at most as
+ *                  many limbs as the modulus \p m.)
```

Comment:
Limbs don't have limbs - what I think is meant is
```suggestion
 * \param p_limbs   The number of limbs of \p p. Must be <= the number of
 *                  limbs in the modulus \p m.)
```